www/js/app.js

```diff
@@ -490,7 +490,6 @@ define(['jquery', 'zimArchiveLoader', 'uiUtil', 'settingsStore','abstractFilesys
             tmpMessageChannel.port1.onmessage = handleMessageChannelMessage;
             // Send the init message to the ServiceWorker, with this MessageChannel as a parameter
             navigator.serviceWorker.controller.postMessage({'action': 'init'}, [tmpMessageChannel.port2]);
-            messageChannel = tmpMessageChannel;
```

Comment:
@ram690514 I don't think this line should be removed. It is part of setting up the Service Worker mode. Have you tested that Service Worker mode (see Configuration) works on this branch even without this?

Reply:
I've been testing this, and I'm not noticing any errors without this line. I'll check the code properly, so don't change it for now.

Reply:
OK, while this does not seem to be needed in Chromium, I don't think it's safe to remove this line without extensive testing, because it is part of a "keepalive" implementation. The SW API may well have been fixed in recent times, but we can't be sure users do not have old browsers, so please restore this line.

Reply:
@Jaifroid : I had restored the element messageChannel and updated the pull request